content/blogposts/2024/top-5-tips-for-using-cody-with-react.md

```diff
@@ -0,0 +1,306 @@
+tags: [blog]
+slug: "top-5-tips-for-using-cody-with-react"
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/og-top-5-tips-for-using-cody-with-react.png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/og-top-5-tips-for-using-cody-with-react.png
+---
+
+<Alert type="secondary">This is a guest blog post from our community. <a href="https://discord.com/servers/sourcegraph-969688426372825169" target="_blank">Join our Discord</a> to meet the author and connect with other developers passionate about Code Search and AI!</Alert>
+
+React is a popular frontend library used for building fast interactive user interfaces. React is a common choice for most organizations in terms of a frontend library. Also, as different LLM models and ChatGPT are here, developers are using them a lot to answer coding questions and write some code. Taking into account how AI helps to boost the productivity of the software developer, many coding AI assistants have been developed, and [Cody](https://about.sourcegraph.com/cody) is one of them created by Sourcegraph. 
+
+In this tutorial, we will build a React app to generate Twitter tweet content using GPT-3 based on the given keyword. We will use Cody as our AI assistant, and learn about five tips to enhance our React development experience.
+
+## Prerequisites
+
+1. [Node.js](https://nodejs.org/en) installed on your machine
+2. Basic knowledge of developing web apps using React
+3. TypeScript (optional)
+4. An OpenAI API key. 
+5. [Cody](https://docs.sourcegraph.com/cody/overview#getting-started) AI assitant installed in your editor. 
+
+## Cody Overview
+
+Cody is an coding AI assistant developed by Sourcegraph which has the complete context of your codebase, and can provide context aware answers. Cody has many features like chat to answer any questions, provide fast useful auto-completions, commands, inline-chat etc. In this tutorial, we will be using these features of Cody to help with our app effectively.
+
+## Top 5 tips for using Cody with React
+
+### 1. Add Enough Context
+
+The first tip while using Cody to keep in mind is always to provide the required context, which is info about the coding question or the bug, to get a helpful response. Cody uses three types of context to give the result keyword context, graph context and the codebase context. Codebase context is used by Cody when the repository is indexed, while in another case, the keyword context is used. For this project, all of the responses provided by Cody will be using the keyword context.
+
+Let's see a few examples of Cody's input prompt, how to provide enough context and build the search bar for our tweet generator app.
+
+The first question I asked Cody is: `I want to build a React app using GPT-4, which can generate the tweet content for me to post on Twitter. Tell me how to begin with this app and how I can develop it in the best possible way.`
```

Comment:
At the top of the article you say we'll use GPT-3, is it GPT-3 or 4 for the app? :)